store.go

```diff
@@ -126,31 +143,13 @@ type Store interface {
 // TxOption holds options for store transactions.
 type TxOption interface{}
 
-type TxOptions []TxOption
-
-type writeLockOption struct {
-}
-
-func (opts TxOptions) InvokeOnRollback() {
-	for _, opt := range opts {
-		if ar, ok := opt.(*OnRollbackOpt); ok {
-			ar.Func()
-		}
-	}
+type WriteLockOption struct {
 }
 
-func (opts TxOptions) InvokeAfterCommit() {
+// Enabled returns whether the WithWriteLock option was specified.
+func (w WriteLockOption) Enabled(opts []TxOption) bool {
```

Comment:
Why is this a method on `WriteLockOption`?

Reply:
I figured it is a better pattern than having a `TxOptions` (alias for a slice) that has all these option-specific functions. Now it's function to read it from an options list is bound to the option, at least.